tests/python_package_test/test_dask.py

```diff
@@ -455,6 +487,74 @@ def test_classifier_pred_contrib(output, task, cluster):
                 assert len(np.unique(preds_with_contrib[:, base_value_col]) == 1)
 
 
+@pytest.mark.parametrize('output', data_output)
+@pytest.mark.parametrize('task', ['binary-classification', 'multiclass-classification'])
+def test_classifier_custom_objective(output, task, cluster):
+    with Client(cluster) as client:
+        X, y, w, _, dX, dy, dw, _ = _create_data(
+            objective=task,
+            output=output
+        )
+
+        params = {
+            "n_estimators": 50,
+            "num_leaves": 31,
+            "min_data": 1,
+            "verbose": -1,
+            "learning_rate": 0.01,
+        }
+
+        if task == 'binary-classification':
+            params.update({
+                'objective': _objective_logistic_regression,
+            })
+        elif task == 'multiclass-classification':
+            params.update({
+                'objective': _objective_logloss,
+                'num_classes': 3
+            })
+
+        dask_classifier = lgb.DaskLGBMClassifier(
+            client=client,
+            time_out=5,
+            tree_learner='data',
+            **params
+        )
+        dask_classifier = dask_classifier.fit(dX, dy, sample_weight=dw)
+        dask_classifier_local = dask_classifier.to_local()
+        p1_proba = dask_classifier.predict_proba(dX).compute()
+        p1_proba_local = dask_classifier_local.predict_proba(X)
+
+        # with a custom objective, predictiion result is a raw score instead of predicted class
+        p1_class = (1.0 / (1.0 + np.exp(-p1_proba))) > 0.5
+        p1_class = p1_class.astype('int64')
+        p1_class_local = (1.0 / (1.0 + np.exp(-p1_proba_local))) > 0.5
+        p1_class_local = p1_class_local.astype('int64')
+
+        local_classifier = lgb.LGBMClassifier(**params)
+        local_classifier.fit(X, y, sample_weight=w)
+        p2_proba = local_classifier.predict_proba(X)
+        p2_class = (1.0 / (1.0 + np.exp(-p1_proba))) > 0.5
+        p2_class = p2_class.astype('int64')
+
+        if task == 'multiclass-classification':
+            p1_class = p1_class.argmax(axis=1)
+            p1_class_local = p1_class_local.argmax(axis=1)
+            p2_class = p2_class.argmax(axis=1)
+
+        # function should have been preserved
+        assert callable(dask_classifier.objective)
+        assert callable(dask_classifier_local.objective)
```

Comment:
I think it's more idiomatic to check post-fit attribute:
https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMModel.html#lightgbm.LGBMModel.objective_
```suggestion
        assert callable(dask_classifier.objective_)
        assert callable(dask_classifier_local.objective_)
```